OpenTrons/opentrons-api
# feat(protocol-designer,-shared-data): add liquid class scaffolding to PD

# Overview

This PR introduces initial work for liquid class support in protocol designer. I create a full liquid class interface based on the API's Pydantic model. I also add utilities for getting all liquid class definitions and a feature flag to hide liquid classes from the UI. Lastly, I add some of the basic functionality for assigning a liquid class to a defined liquid, and displaying that liquid class in LiquidCard. A liquid class assigned to a defined liquid will be accessible through extended existing selectors (allIngredientNamesIds) for step forms.

closes AUTH-965
closes AUTH-968

## Test Plan and Hands on Testing

- navigate to settings and flip on liquid classes feature flag
- navigate to protocol overview and add or modify a liquid
- select a liquid class (should only see water as this is the only built out definition as of now)
- save and verify that liquid definitions reflect liquid class
- add this liquid to a labware or inspect a labware that has already loaded this liquid
- verify that liquid card in toolbox reflects liquid class

https://github.com/user-attachments/assets/48990650-b3b0-483d-9ea3-8f2dec00f3af

## Changelog

- add liquid class interface
- add liquid class feature flag
- add liquid class definitions getter
- update selectors to accommodate liquid classes
- dad liquid class feature to DefineLiquidsModal, LiquidsToolbox, LiquidCard, LiquidDefinitions (behind feature flag)
- update tests

## Review requests

- see test plan

## Risk assessment

medium
